.ci/Jenkinsfile

```diff
@@ -160,14 +166,42 @@ def generateTestCommandStage(Map args = [:]){
           }
 
           if (publishCoverage) {
-            coverageReport('build/test-coverage')
+            stashCoverageReport()
           }
         }
       }
     }
   }
 }
 
+def getCoverageBucketURI() {
+  return "gs://${JOB_GCS_BUCKET}/" + getCoveragePathPrefix()
+}
+
+def getCoveragePathPrefix() {
+  return "${env.JOB_NAME}-${env.BUILD_ID}/test-coverage/"
+}
+
+def publishCoverageReports() {
+  stage('Publish coverage reports') {
+    dir("${BASE_DIR}") {
+      def bucketUri = getCoverageBucketURI() + "*.xml"
+      googleStorageDownload(bucketUri: bucketUri, credentialsId: "${JOB_GCS_CREDENTIALS}", localDirectory: 'build/test-coverage', pathPrefix: getCoveragePathPrefix())
+      coverageReport('build/test-coverage')
+    }
+  }
+}
+
+def stashCoverageReport() {
```

Comment:
Pushing to a GCP allows to publish the code coverage report just once.

@v1v, I believe this set of helpers can be wrapped into a new step, as I hard-copied them from the integrations pipeline.